x/oracle/keeper/oracle.go

```diff
@@ -42,6 +43,53 @@ func (k Keeper) RegisterOracle(ctx sdk.Context, msg *types.MsgRegisterOracle) er
 	return nil
 }
 
+func (k Keeper) UpdateOracleInfo(ctx sdk.Context, msg *types.MsgUpdateOracleInfo) error {
+	oracle, err := k.GetOracle(ctx, msg.OracleAddress)
+	if err != nil {
+		return sdkerrors.Wrapf(types.ErrUpdateOracle, err.Error())
+	}
+
+	blockTime := ctx.BlockHeader().Time
+
+	if !oracle.OracleCommissionRate.Equal(msg.OracleCommissionRate) {
+		if err := k.ValidateOracleCommission(oracle, blockTime, msg.OracleCommissionRate); err != nil {
+			return sdkerrors.Wrapf(types.ErrUpdateOracle, err.Error())
+		}
+	}
+
+	oracle.UpdateTime = blockTime
+	oracle.Endpoint = msg.Endpoint
```

Comment:
Why don't we update the endpoint only when the length of `msg.Endpoint` is greater than 0?
In other words, there is no need to enter endpoint when user want to change commission rate only.
```suggestion
        if len(msg.Endpoint) > 0 {
            oracle.Endpoint = msg.Endpoint
        }
```

Reply:
I missed this part. thank you